autopush-common/src/db/mod.rs

```diff
@@ -336,74 +336,100 @@ impl DynamoStorage {
-                if !resp.messages.is_empty() {
-                    debug!("Topic message returns: {:?}", resp.messages);
-                    return future::Either::Left(future::ok(CheckStorageResponse {
-                        include_topic: true,
-                        messages: resp.messages,
-                        timestamp: resp.timestamp,
-                    }));
-                }
-                // Use the timestamp returned by the topic query if we were looking at the topics
-                let timestamp = if include_topic {
-                    resp.timestamp
-                } else {
-                    timestamp
-                };
-                future::Either::Right(
-                    {
-                        if resp.messages.is_empty() || resp.timestamp.is_some() {
-                            future::Either::Left(commands::fetch_timestamp_messages(
-                                ddb,
-                                &metrics,
-                                table_name.as_ref(),
-                                &uaid,
-                                timestamp,
-                                10 as u32,
-                            ))
-                        } else {
-                            future::Either::Right(future::ok(Default::default()))
-                        }
-                    }
-                    .and_then(move |resp: FetchMessageResponse| {
-                        // If we didn't get a timestamp off the last query, use the original
-                        // value if passed one
-                        let timestamp = resp.timestamp.or(timestamp);
-                        future::Either::Right(future::ok(CheckStorageResponse {
-                            include_topic: false,
-                            messages: resp.messages,
-                            timestamp,
-                        }))
-                    }),
-                )
-            }),
-        )
+        }
+        .and_then(move |resp: FetchMessageResponse| {
+            // If we didn't get a timestamp off the last query, use the original
+            // value if passed one
+            let timestamp = resp.timestamp.or(timestamp);
+            future::ok(CheckStorageResponse {
+                include_topic: false,
+                messages: resp.messages,
+                timestamp,
+            })
+        }))
+    }
+
+    pub async fn check_storage(
+        &self,
+        table_name: &str,
+        uaid: &Uuid,
+        include_topic: bool,
+        timestamp: Option<u64>,
+    ) -> LocalBoxFuture<'static, Result<CheckStorageResponse>> {
+        let uaid = *uaid;
+        let table_name = table_name.clone().to_string();
+
+        let resp = self.check_include_topic(
+            include_topic,
+            table_name.clone(),
+            &uaid
+        ).await;
+        self.check_if_empty(table_name, &uaid, include_topic, timestamp, resp)
```

Comment:
And this needs to change to `self.check_if_empty(table_name, &uaid, include_topic, timestamp, resp).await`